examples/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.11)
+
+get_filename_component(ImGui_SRCDIR "${CMAKE_CURRENT_LIST_DIR}" DIRECTORY)
+
+include("${CMAKE_CURRENT_LIST_DIR}/ImGuiModule.cmake")
+
+set(ImGui_OPTIONS)
+
+imgui_option(EXAMPLES "Dear ImGui example applications" ON)
+imgui_option(BINDINGS "Dear ImGui example bindings" ON)
+imgui_option(MISC     "Dear ImGui misc features" ON)
+imgui_option(3RDPARTY "Dear ImGui example dependencies" ON)
+
+file(GENERATE OUTPUT ImGuiOptions.cmake
+    CONTENT "${ImGui_OPTIONS_CMAKE}")
+
+project(imgui_examples
+    LANGUAGES CXX)
+
+include("${CMAKE_CURRENT_LIST_DIR}/ImGuiTargets.cmake")
+
+set(ImGui_DIR "${CMAKE_CURRENT_LIST_DIR}")
+
+imgui_example(example_null
```

Comment:
It will be nice to only have this example to be built by default and then have imgui_option for building each of the examples, because right now I can't imagine a case where user would want to build *every* example - this requires too much dependencies.

Reply:
Usually everything is built by default and then for people that want to tailor their builds they manually set options to `OFF` to exclude things to be built.

I heavily suggest this be the case, as fresh clones for contributing won't need to guess at what options to build just to e.g. fix a bug or add a feature, whereas a consuming project will be willing to take the time to tailor the builds to their usecases.

tl;dr everything should be enabled by default (a full build).

Reply:
I don't think that applies here, as you are usually only interested in one particular combination of platform and render backend. 

I'd wager a guess that contributors are more likely to figure out what build options to use (or spend the time to download all the different bindings) than people who "just want to try it out". That is double true for people that don't use ImGui directly, but where ImGui is a dependency of their dependency.

Reply:
@qix- Actually, the modern convention is to only default everything on when `"${CMAKE_SOURCE_DIR}" STREQUAL "${CMAKE_CURRENT_SOURCE_DIR}"` as a way of telling whether you are standalone.

Also, I find that packages with lots of examples/testing/ci options are easier to work with when their "build this thing?" options are specified in the nearest cmakelists.txt to the particular example/package: in a setup like dear imguis, where the examples double as documentation, the easier it is to simply lift an example directory and start coding, the better. I know it adds some maintenance headache, but welcome to library maintenance.